node/src/accountant/mod.rs

```diff
@@ -41,46 +41,43 @@ use actix::Context;
 use actix::Handler;
 use actix::Message;
 use actix::Recipient;
-use itertools::Itertools;
 use masq_lib::crash_point::CrashPoint;
 use masq_lib::logger::Logger;
 use masq_lib::messages::UiFinancialsResponse;
 use masq_lib::messages::{FromMessageBody, ToMessageBody, UiFinancialsRequest};
 use masq_lib::ui_gateway::MessageTarget::ClientId;
 use masq_lib::ui_gateway::{NodeFromUiMessage, NodeToUiMessage};
-use masq_lib::utils::{plus, ExpectValue};
+use masq_lib::utils::ExpectValue;
 use payable_dao::PayableDao;
 use receivable_dao::ReceivableDao;
-#[cfg(test)]
-use std::any::Any;
 use std::default::Default;
-use std::ops::Add;
 use std::path::Path;
-use std::time::{Duration, SystemTime};
+use std::rc::Rc;
+use std::time::SystemTime;
 use web3::types::{TransactionReceipt, H256};
 
 pub const CRASH_KEY: &str = "ACCOUNTANT";
 
 pub const DEFAULT_PENDING_TOO_LONG_SEC: u64 = 21_600; //6 hours
 
 pub struct Accountant {
-    config: AccountantConfig,
+    scan_intervals: ScanIntervals,
+    suppress_initial_scans_opt: Option<bool>,
```

Comment:
Important thing to note here is that the field of `suppress_initial_scans_opt` is initialised in Neighborhood's constructor, and this field is used (first and only usage) in the `handle()` of `StartMessage`.

If we use an `Option<T>` over here, then we can _panic_ in case the field is intitalized.

Hence, it acts as the safety measurement, which prevents the StartMessage to get processed before the initialization of Neighbourhood. This was the main reason why wrapping it with an `Option<T>` was important.

Reply:
Okay. I looked at it (although my IDE didn't work right at the moment and so I couldn't pursue all details) but my opinion is:

1)  `suppress_initial_scans_opt` is purely a user based value. It is defined since the moment of parsing inputs. 

2) All actors, without an exception, are built up in the ActorSystemFactory and we can say that all of them at the same time, in the same big function. In other words, Neighborhood lives since the same time as Accountant does. 

3) The value for `suppress_initial_scans_opt` is tempting to be just `false` most of the time. Put differently, true is a very special case, maybe for testing at the end of the day. Therefore we know easily what would be the best default: just `false`. 

4) When `BootstrapperConfig` is being filled with data, it's natural for us to fill it with all possible values we know and then use them whenever they are needed at the startup of individual actors. We usually approach a constructor of a certain actor and we supply values which become the base values inside the body of the actor.

5) the optionality you have with this field in Accountant might be misleading. One could think of it that it will be decided later in the life of the program what value it's gonna be. However, we know it since the startup. Using just a boolean is simple enough and looks very clear. 

6) I don't like the idea of removing the value and leaving a None there for the rest of the Node's life. And because values like `None` and `false` will probably take up the same size of memory. I don't see an argument for None there either.

Well... I'm just trying to say that I don't see a huge solid argument for what I've tried to talk you into but when I weight all the smaller points I made I feel like I believe in the right thing. Hopefully you will think it over again.